src/gui/sdl_mapper.cpp

```diff
@@ -2983,22 +3057,9 @@ static void CreateBindGroups() {
 		return;
 
 	if (joytype != JOY_NONE_FOUND) {
-#if defined (REDUCE_JOYSTICK_POLLING)
-		// direct access to the SDL joystick, thus removed from the event handling
-		if (mapper.sticks.num)
-			SDL_JoystickEventState(SDL_DISABLE);
-#else
 		// enable joystick event handling
 		if (mapper.sticks.num)
 			SDL_JoystickEventState(SDL_ENABLE);
```

Comment:
This disabling and re-enabling joystick events is a mess.  Glad you got rid of most of it.  I honestly think we should never be calling this function.  This is from `SDL_joystick.h`.  Note the big ole warning at the bottom:

```c
 * Enable/disable joystick event polling.
 *
 * If joystick events are disabled, you must call SDL_JoystickUpdate()
 * yourself and manually check the state of the joystick when you want
 * joystick information.
 *
 * It is recommended that you leave joystick event handling enabled.
 *
 * **WARNING**: Calling this function may delete all events currently in SDL's
 * event queue.
 ```

As I understand it, we're polling the joysticks rather than using SDL events for button presses and such.  However, we need joystick events for hotplug to work (otherwise how would we know if something got plugged/unplugged).  If we get button press events we don't care about, we can just ignore them and that's fine.

This is the only place we're calling this function but events should be enabled by default.  I'd suggest removing this so we don't accidentally purge our entire event queue.

Reply:
Agreed, fixed.